optuna/integration/sklearn.py

```diff
@@ -55,30 +47,53 @@
     from typing import Optional  # NOQA
     from typing import Union  # NOQA
 
+    ArrayLikeType = Union[List, np.ndarray, pd.Series]
     OneDimArrayLikeType = Union[List[float], np.ndarray, pd.Series]
     TwoDimArrayLikeType = \
         Union[List[List[float]], np.ndarray, pd.DataFrame, spmatrix]
 
 _logger = logging.get_logger(__name__)
 
 
+def _is_arraylike(x):
+    # type: (Any) -> bool
+
+    return (
+        hasattr(x, '__len__') or
+        hasattr(x, 'shape') or
+        hasattr(x, '__array__')
+    )
+
+
+def _num_samples(x):
+    # type: (ArrayLikeType) -> int
+
+    try:
+        return len(x)
+    except TypeError:
+        raise TypeError('Expected sequence or array-like, got %s' % type(x))
+
+
 def _check_fit_params(
     X,  # type: TwoDimArrayLikeType
     fit_params,  # type: Dict
     indices  # type: OneDimArrayLikeType
 ):
     # type: (...) -> Dict
 
-    if _sklearn_version >= '0.22.1':
-        return _sklearn_check_fit_params(X, fit_params, indices)
-    else:  # '_sklearn_version < 0.22.1'
-        return {
-            key: _sklearn_index_param_value(
-                X,
-                value,
-                indices
-            ) for key, value in fit_params.items()
-        }
+    fit_params_validated = {}
+    for key, value in fit_params.items():
+        if (
+            not _is_arraylike(value) or
+            _num_samples(value) != _num_samples(X)
+        ):
+            fit_params_validated[key] = value
+        else:
+            fit_params_validated[key] = value
```

Comment:
The original code [here](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/validation.py#L1332) applies the `_make_indexiable` to `value`. Do we have any drawbacks?

Reply:
Sorry, you're right. I missed it.
I added `_make_indexable` in 2be88c7.